boa_engine/src/builtins/promise/promise_job.rs

```diff
@@ -0,0 +1,194 @@
+use gc::{Finalize, Trace};
+
+use crate::{
+    builtins::promise::{ReactionRecord, ReactionType},
+    job::JobCallback,
+    object::{FunctionBuilder, JsObject},
+    Context, JsValue,
+};
+
+use super::{Promise, PromiseCapability};
+
+pub(crate) struct PromiseJob;
```

Comment:
Could we add some documentation here, add the reference to the spec, and derive `Debug`, `Clone` and `Copy` for it?